photon-api/src/main/scala/com/linkedin/photon/ml/data/RandomEffectDataSet.scala

```diff
@@ -73,15 +78,29 @@ protected[ml] class RandomEffectDataSet(
     update(updatedActiveData, updatedPassiveDataOption)
   }
 
+  /**
+   *
+   * @return The Spark context
+   */
   override def sparkContext: SparkContext = activeData.sparkContext
```

Comment:
I'll make small corrections, and I'll document active/passive data, but that's roughly as far as I'm gonna go in this PR - I don't want an already huge PR to get bogged down also fixing our documentation.